types/parser_driver/value_expr.go

```diff
@@ -173,7 +173,8 @@ func newValueExpr(value interface{}, charset string, collate string) ast.ValueEx
 		return ve
 	}
 	ve := &ValueExpr{}
-	ve.SetValue(value)
+	ve.Type = types.FieldType{Charset: charset, Collate: collate}
```

Comment:
The charset and collate here is used for string. Besides, the newValueExpr in parser also needs to change.

Reply:
We do not need to check string/enum/set here.
1. Charset and collate are used for string, set, enum.
2. ve.Datum.SetValue only set the specified charset and collate for these 3 types.
3. types.DefaultTypeForValue will use default charset and collate if the type is not string/enum/set.

Reply:
And we do not need to change the implementation for parser now.
The interface of Datum in parser is not the same as tidb.